app/js/disease-associations-controller.js

```diff
@@ -150,12 +164,123 @@ angular.module('cttvControllers')
+    render(cttvLocationState.getState(), cttvLocationState.getOldState());
 
+    $scope.uploadedFile = function (element) {
+        $scope.$apply(function ($scope) {
+            $scope.files = element.files;
+        });
 
-    cttvUtils.clearErrors();
-    $scope.filters = cttvLocationState.getState()[facetsId] || {} ;
-    render( cttvLocationState.getState(), cttvLocationState.getOldState() );
+        $scope.addFile();
+    };
+
+    $scope.removeTargets = function(){
+        var theElement = document.getElementById("myFileInput");
+
+        theElement.value = null;
+        initFilterByFile();
+        //console.log("removeTargets:$scope.files=", $scope.files );
+        getFacets($scope.filters);
+    };
+
+    $scope.addFile = function () {
+        $scope.validateFile($scope.files[0]);
+    };
+
+    $scope.validateFile = function (file) {
+        $scope.fileName = file.name;
+        var reader = new FileReader();
+        reader.onloadend = function (evt) {
+            //do something with file content here
+            var myFileContent = evt.target.result;
+            $scope.targetNameArray = myFileContent.replace(/(\r\n|\n|\r)/gm, '\n').split('\n');
+            $scope.targetNameArray = $scope.targetNameArray.filter(function(e){ return e.trim();}); //get rid of empty strings
+
+            getTargetIds();
+        };
+        reader.readAsText(file);
+    };
+
+    var getTargetIds = function() {
+        var promise = $q(function (resolve) {
+            resolve("");
+        });
+
+        $scope.targetNameArray.forEach(function (targetName){
+            promise = promise.then(function() {
+               return getTargetId(targetName);
+           });
+        });
+
+        promise.then(function (res) {
+            $log.log("PROMISE");
+            $scope.targetArray = $scope.targetIdArray;
+            $scope.excludedTargetArray = $scope.targetNameIdDict.filter(function(e){return !e.id;});
+            $scope.fuzzyTargetArray = $scope.targetNameIdDict.filter(function(e){return e.name.localeCompare(e.label) !== 0 && e.id.localeCompare(e.name) !== 0;});
+
+            $log.log("123:targetNameIdDict", $scope.targetNameIdDict);
+            $log.log("123:excludedTargetArray", $scope.excludedTargetArray);
+            $log.log("123:fuzzyTargetArray", $scope.fuzzyTargetArray);
+            getFacets($scope.filters);
+        });
 
+    };
+
+    var getExcludedOnes =  function(item){
+        //console.log("getExcludedOnes:item=", item);
+        var isIdEmpty = false;
+        //console.log("getExcludedOnes:item.id=", item.id);
+        if (item.id === ''){
+            //console.log("getExcludedOnes:item.id==''", true);
+            isIdEmpty = true;
+        }
+        if (item.id === ""){
+            //console.log("getExcludedOnes:item.id==str", true);
+            isIdEmpty = true;
+        }
+        return isIdEmpty;
+    };
 
+    var getTargetId = function (targetName) {
+        if (targetName.startsWith("ENSG")){
```

Comment:
why do you need this?
I think if you just fire searches with whatever name is passed (id or symbol) the search will treat them equally and it is good to fire the searches with ensembl ids anyway